blocks/main_loop.py

```diff
@@ -233,8 +234,20 @@ def _run_iteration(self):
         return True
 
     def _run_extensions(self, method_name, *args):
+        class Callback(str):
+            def __eq__(self, other):
+                callbackname = [key for key, value
+                                in TrainingExtension.__dict__.items()
+                                if getattr(value, '_is_callback', False)]
+                if other not in callbackname:
+                    if isinstance(other, str):
+                        raise ValueError(str(other) +
```

Comment:
It's better to use Python's new string formatting, so `"{} is not a valid callback".format(other)`. The casting to a string will happen automatically.